v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/utils/ElasticsearchIO.java

```diff
@@ -414,6 +422,29 @@ public ConnectionConfiguration withTrustSelfSignedCerts(boolean trustSelfSignedC
       return builder().setTrustSelfSignedCerts(trustSelfSignedCerts).build();
     }
 
+    /**
+     * Sets User-Agent headers.
+     *
+     * <p>Sets Authorization: ApiKey if ApiKey is present Sets Authorization: Bearer if BearerToken
+     * is present
+     *
+     * @return an array of {@link Header} that apply to all HTTP requests
+     */
+    protected Header[] configureDefaultHeaders() {
+      final ArrayList<Header> headerList = new ArrayList<Header>();
+
+      headerList.add(new BasicHeader("user-agent", getUserAgent()));
```

Comment:
```suggestion
      headerList.add(new BasicHeader("User-Agent", getUserAgent()));
```
Nit. Casing _shouldn't_ matter, but I believe this shouldn't be all lowercase.

Reply:
HTTP headers are case-insensitive but I can change it because capitalised name looks better :).